codec-redis/src/main/java/io/netty/handler/codec/redis/FixedRedisMessagePool.java

```diff
@@ -0,0 +1,146 @@
+     * Creates a {@link FixedRedisMessagePool} instance.
+     */
+    private FixedRedisMessagePool() {
+        byteBufToSimpleStrings = new HashMap<ByteBuf, SimpleStringRedisMessage>(DEFAULT_SIMPLE_STRINGS.length);
+        stringToSimpleStrings = new HashMap<String, SimpleStringRedisMessage>(DEFAULT_SIMPLE_STRINGS.length);
+        for (String message : DEFAULT_SIMPLE_STRINGS) {
+            ByteBuf key = Unpooled.unmodifiableBuffer(
+                    Unpooled.unreleasableBuffer(Unpooled.wrappedBuffer(message.getBytes(CharsetUtil.UTF_8))));
+            SimpleStringRedisMessage cached = new SimpleStringRedisMessage(message);
+            byteBufToSimpleStrings.put(key, cached);
+            stringToSimpleStrings.put(message, cached);
+        }
+
+        byteBufToErrors = new HashMap<ByteBuf, ErrorRedisMessage>(DEFAULT_ERRORS.length);
+        stringToErrors = new HashMap<String, ErrorRedisMessage>(DEFAULT_ERRORS.length);
+        for (String message : DEFAULT_ERRORS) {
+            ByteBuf key = Unpooled.unmodifiableBuffer(
+                    Unpooled.unreleasableBuffer(Unpooled.wrappedBuffer(message.getBytes(CharsetUtil.UTF_8))));
+            ErrorRedisMessage cached = new ErrorRedisMessage(message);
+            byteBufToErrors.put(key, cached);
+            stringToErrors.put(message, cached);
+        }
+
+        byteBufToIntegers = new HashMap<ByteBuf, IntegerRedisMessage>();
+        longToIntegers = new LongObjectHashMap<IntegerRedisMessage>();
+        longToByteBufs = new LongObjectHashMap<ByteBuf>();
+        for (long value = 0; value < 128; value++) {
+            ByteBuf key = Unpooled.unmodifiableBuffer(Unpooled.unreleasableBuffer(
+                    Unpooled.wrappedBuffer(Long.toString(value).getBytes(CharsetUtil.US_ASCII))));
+            IntegerRedisMessage cached = new IntegerRedisMessage(value);
+            byteBufToIntegers.put(key, cached);
+            longToIntegers.put(value, cached);
+            longToByteBufs.put(value, key);
+        }
+    }
+
+    @Override
+    public SimpleStringRedisMessage getSimpleString(String content) {
+        return stringToSimpleStrings.get(content);
+    }
+
+    @Override
+    public SimpleStringRedisMessage getSimpleString(ByteBuf content) {
+        return byteBufToSimpleStrings.get(content);
+    }
+
+    @Override
+    public ErrorRedisMessage getError(String content) {
+        return stringToErrors.get(content);
+    }
+
+    @Override
+    public ErrorRedisMessage getError(ByteBuf content) {
+        return byteBufToErrors.get(content);
+    }
+
+    @Override
+    public IntegerRedisMessage getInteger(long value) {
+        return longToIntegers.get(value);
+    }
+
+    @Override
+    public IntegerRedisMessage getInteger(ByteBuf content) {
+        return byteBufToIntegers.get(content);
+    }
+
+    @Override
+    public ByteBuf getByteBufOfInteger(long value) {
+        ByteBuf buf = longToByteBufs.get(value);
+        return buf != null ? buf.slice() : null;
```

Comment:
It may be best to change the signature of this method so we don't have to return a `ByteBuf` every time and instead can write directly into a `ByteBuf`. Considering what we assume to be the "most common" cached values will only be 3 characters ... creating a slice is relatively high overhead.  This will give the cache more freedom to represent the small values in a more efficient manner.

```java
public void writeAsciiInteger(ByteBuf buf, long value) {
}

public ByteBuf writeAsciiInteger(ByteBufAllocator alloc, long value) {
  // default implementation, but caches can allocate "smaller" buffers if the want to.
  ByteBuf buf = alloc.bufer(20);
  writeAsciiInteger(buf, value);
  return buf;
}
```